src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -1035,6 +1043,8 @@ public Collection<Object> createComponents(
 
         final XFFResolver xffResolver = new XFFResolver(threadPool);
         backendRegistry = new BackendRegistry(settings, adminDns, xffResolver, auditLog, threadPool);
+        subject.setThreadContext(threadPool.getThreadContext());
```

Comment:
This looks like an implementation detail of a Subject in this repo. Would it make sense to make this a constructor arg?

Reply:
Why duplicated the whole thread context here?

Reply:
I've swapped this for the NoopSubject for the moment, until we are using it lets not insert anymore context than necessary into this the IdentityPlugin implementation